play-mailer-guice/src/test/scala/play/api/libs/mailer/MailerPluginGuiceSpec.scala

```diff
@@ -49,7 +50,8 @@ class MailerPluginGuiceSpec extends Specification with Mockito {
       there was two(mockedConfigurationProvider).get()
     }
     "validate the configuration" in new WithApplication(applicationWithMoreMailerConfiguration) {
-      app.injector.instanceOf(bind[SMTPConfiguration]) must ===(SMTPConfiguration("typesafe.org", 25, user = Some("typesafe"), password = Some("typesafe")))
+      app.injector.instanceOf(bind[SMTPConfiguration]) must ===(SMTPConfiguration("typesafe.org", 25,
+        user = Some("typesafe"), password = Some("typesafe"), props = ConfigFactory.parseString("ssl.checkserveridentity=true")))
```

Comment:
So when constructing the `SMTPConfiguration` from code rather than from configuration the default is still off. That might still be a risk.

Reply:
Probably the best we can do is add a log WARN when that happens.

```scala
case class SMTPConfiguration(host: String,
                             port: Int,
                             ssl: Boolean = false,
                             tls: Boolean = false,
                             tlsRequired: Boolean = false,
                             user: Option[String] = None,
                             password: Option[String] = None,
                             debugMode: Boolean = false,
                             timeout: Option[Int] = None,
                             connectionTimeout: Option[Int] = None,
                             props: Config = ConfigFactory.empty(),
                             mock: Boolean = false) {
  private val logger: Logger =
    LoggerFactory.getLogger(classOf[SMTPConfiguration])

  if ((ssl || tls) && !props.getBoolean(
        "ssl.checkserveridentity"
      ))
    logger.warn(
      "You are using ssl/tls but no hostname verification. Enable hostname verification using play.mailer.props.ssl.checkserveridentity = true"
    )
}
```

Reply:
I think it's fine as is, no one loads one of this from code, if you're going to all that effort, I don't know why you'd even use this library, it doesn't do much.